e2e-tests/tests/test_jolteon_consensus_rpc.py

```diff
@@ -0,0 +1,424 @@
+            assert False, "Failed to get consensus state from all endpoints"
+        
+        replica_state = replica_state_result['result']
+        round_info = round_info_result['result']
+        qc = qc_result['result']
+        
+        logger.info(f"Replica state: {replica_state}")
+        logger.info(f"Round info: {round_info}")
+        logger.info(f"QC: {qc}")
+        
+        # Verify round consistency between endpoints
+        assert replica_state['r_curr'] == round_info['r_curr'], \
+            f"Current round mismatch: replica_state={replica_state['r_curr']}, round_info={round_info['r_curr']}"
+        
+        assert replica_state['r_vote'] == round_info['r_vote'], \
+            f"Voted round mismatch: replica_state={replica_state['r_vote']}, round_info={round_info['r_vote']}"
+        
+        assert replica_state['r_lock'] == round_info['r_lock'], \
+            f"Locked round mismatch: replica_state={replica_state['r_lock']}, round_info={round_info['r_lock']}"
+        
+        # Verify QC consistency
+        assert replica_state['qc_high']['round'] == qc['round'], \
+            f"QC round mismatch: replica_state={replica_state['qc_high']['round']}, qc={qc['round']}"
+        
+        assert replica_state['qc_high']['block_hash'] == qc['block_hash'], \
+            f"QC block hash mismatch: replica_state={replica_state['qc_high']['block_hash']}, qc={qc['block_hash']}"
+        
+        assert replica_state['qc_high']['vote_count'] == qc['vote_count'], \
+            f"QC vote count mismatch: replica_state={replica_state['qc_high']['vote_count']}, qc={qc['vote_count']}"
+        
+        # Verify logical consistency
+        assert replica_state['r_curr'] >= replica_state['r_vote'], \
+            f"Current round should be >= voted round: {replica_state['r_curr']} < {replica_state['r_vote']}"
+        
+        assert replica_state['r_vote'] >= replica_state['r_lock'], \
+            f"Voted round should be >= locked round: {replica_state['r_vote']} < {replica_state['r_lock']}"
+        
+        assert replica_state['qc_high']['round'] <= replica_state['r_curr'], \
+            f"QC round should be <= current round: {replica_state['qc_high']['round']} > {replica_state['r_curr']}"
+        
+        logger.info("✅ Consensus state consistency test passed")
+
+    @mark.test_key('JOLTEON-RPC-006')
+    def test_jolteon_safety_properties(self, api: BlockchainApi, config: ApiConfig):
+        """Test Jolteon safety properties using RPC data
+        
+        This test verifies fundamental safety guarantees:
+        - Round monotonicity
+        - QC progression
+        - Lock consistency
+        """
+        logger.info("Starting Jolteon safety properties test")
+        
+        # Monitor consensus state over time
+        wait_time = 60  # Monitor for 1 minute
+        check_interval = 15  # Check every 15 seconds
+        logger.info(f"Monitoring consensus state for {wait_time} seconds...")
```

Comment:
Let's get rid of magic numbers please. 
If we can't get exactly when the next block is incoming at least let's wait a protocol value period, such as round_timeout.